pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -1146,12 +1145,11 @@ func LoadIPSecKeys(log *slog.Logger, r io.Reader) (int, uint8, error) {
 		ipSecKey.Spi = spi
 		ipSecKey.ESN = esn
 
-		if ipSecKeysGlobal[""] != nil {
-			oldSpi = ipSecKeysGlobal[""].Spi
+		if oldKey, ok := ipSecKeysGlobal[""]; ok {
+			// TODO(smagnani96) add checks to return error if unchanged SPI
+			ipSecKeysRemovalTime[oldKey.Spi] = time.Now()
 		}
 		ipSecKeysGlobal[""] = ipSecKey
-
-		ipSecKeysRemovalTime[oldSpi] = time.Now()
```

Comment:
What's the impact of setting this for `oldSpi=0`? Could it lead to us incorrectly deleting XFRM policies that do not have an SPI encoded in the mark? If so, how come we didn't hit this bug before?

Reply:
AFAIU, it doesn't lead to an incorrect deletion of policies w/o an SPI. The following line should prevent that https://github.com/cilium/cilium/blob/main/pkg/datapath/linux/ipsec/ipsec_linux.go#L1356.

Reply:
Ah, right! Good :relieved: 

Could you clarify that in the commit description (that the "bug" doesn't have any user-visible impact and why)?